nautobot/dcim/models/locations.py

```diff
@@ -199,6 +199,20 @@ def base_site(self):
         """The site that this Location belongs to, if any, or that its root ancestor belongs to, if any."""
         return self.site or self.ancestors().first().site
 
+    @property
+    def display(self):
+        # prefix with parent name allows location name identity if there are the same name exists under different parents.
+        if self.name:
+            if self.parent:
+                if self.parent.parent:
+                    return f"[{self.parent.parent}->{self.parent}] {self.name}"
+                else:
+                    return f"[{self.parent}] {self.name}"
+            else:
+                return self.name
+        else:
+            return ""
```

Comment:
I can expect that there will be cases where 2 levels of parentage is considered insufficient to uniquely identify a location. Maybe we can do something like:

```python
def display(self):
    display_str = self.name
    parent = self.parent
    for i in range(10):  # "10 levels should be more than enough for anybody"
        if parent is None:
            break
        display_str = f"{parent.name} → {display_str}"
        parent = parent.parent
    return display_str
```

Reply:
I'm worried about the recursion impact if we go too far with too long of a list. I would guess Django will have to make an individual SQL call to get the next parent. Considering we are looking to ditch cacheops it may make repeated calls for the same more-root parents.

Reply:
@glennmatthews the result looks great. didn't see too much performance issue. perhaps. I am on local?
![image](https://user-images.githubusercontent.com/110844839/188659416-54c7b920-fc4b-4538-8b7f-83eeeb3d46a9.png)

Reply:
Let's check and make sure that both the drop-down and rendered field are OK display-wise when we have a highly nested hierarchy and long names (perhaps `Campus` -> `Building` -> `Floor` -> `Room` as location-types, with locations `Raleigh-Durham Primary Campus` -> `1234 West Really-Long-Street-Name Avenue` -> `Thirteenth Floor` -> `Room 1301`, just as an example)